src/validators/tuple.rs

```diff
@@ -0,0 +1,220 @@
+                        Err(ValError::LineErrors(line_errors)) => {
+                            let loc = vec![LocItem::I(index)];
+                            errors.extend(line_errors.into_iter().map(|err| err.with_prefix_location(&loc)));
+                        }
+                        Err(err) => return Err(err),
+                    }
+                }
+                if errors.is_empty() {
+                    Ok(PyTuple::new(py, &output).into_py(py))
+                } else {
+                    Err(ValError::LineErrors(errors))
+                }
+            }
+            None => {
+                let output: Vec<PyObject> = tuple.generic_iter().map(|(_, item)| item.to_py(py)).collect();
+                Ok(PyTuple::new(py, &output).into_py(py))
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct TupleFixLenValidator {
+    strict: bool,
+    items_validators: Vec<CombinedValidator>,
+}
+
+impl BuildValidator for TupleFixLenValidator {
+    const EXPECTED_TYPE: &'static str = "tuple-fix-len";
+
+    fn build(
+        schema: &PyDict,
+        config: Option<&PyDict>,
+        build_context: &mut BuildContext,
+    ) -> PyResult<CombinedValidator> {
+        let items: &PyList = schema.get_as_req("items")?;
+        if items.is_empty() {
+            return py_error!("Missing schemas for tuple elements");
+        }
+
+        Ok(Self {
+            strict: is_strict(schema, config)?,
+            items_validators: items
+                .iter()
+                .map(|item| build_validator(item, config, build_context).unwrap().0)
+                .collect(),
+        }
+        .into())
+    }
+}
+
+impl Validator for TupleFixLenValidator {
+    fn validate<'s, 'data>(
+        &'s self,
+        py: Python<'data>,
+        input: &'data dyn Input,
+        extra: &Extra,
+        slots: &'data [CombinedValidator],
+    ) -> ValResult<'data, PyObject> {
+        let tuple = match self.strict {
+            true => input.strict_tuple()?,
+            false => input.lax_tuple()?,
+        };
+        self._validation_logic(py, input, tuple, extra, slots)
+    }
+
+    fn validate_strict<'s, 'data>(
+        &'s self,
+        py: Python<'data>,
+        input: &'data dyn Input,
+        extra: &Extra,
+        slots: &'data [CombinedValidator],
+    ) -> ValResult<'data, PyObject> {
+        self._validation_logic(py, input, input.strict_tuple()?, extra, slots)
+    }
+
+    fn get_name(&self, _py: Python) -> String {
+        Self::EXPECTED_TYPE.to_string()
+    }
+}
+
+impl TupleFixLenValidator {
+    fn _validation_logic<'s, 'data>(
+        &'s self,
+        py: Python<'data>,
+        _input: &'data dyn Input,
+        tuple: GenericSequence<'data>,
+        extra: &Extra,
+        slots: &'data [CombinedValidator],
+    ) -> ValResult<'data, PyObject> {
+        let length = tuple.generic_len();
+        let mut output: Vec<PyObject> = Vec::with_capacity(length);
```

Comment:
```suggestion
        let mut output: Vec<PyObject> = Vec::with_capacity(self.item_validators.len());
```

`output` should always have the same length as `items_validators`, not necessarily the same as the input.